pytest_watch/watcher.py

```diff
@@ -92,8 +94,8 @@ def run(self, summary=None):
             print(STYLE_NORMAL + msg + Fore.RESET + Style.NORMAL)
         if self.beforerun:
             os.system(self.beforerun)
-        exit_code = subprocess.call(['py.test'] + self.args,
-                                    shell=(sys.platform == 'win32'))
+        exit_code = pytest.main(self.args, plugins=[ReportCollectorPlugin()])
```

Comment:
Unfortunately, I don't think this will work. Try this for yourself, but last time I tried to use `pytest.main` instead of subprocess, code updates weren't reflected in the tests. I'm guessing it's because pytest is importing your tests and therefore application code, and so further calls to `pytest.main` uses the in-memory modules instead of re-importing.

Instead of attempting to bust cached modules, using subprocess guarantees that `py.test` is working with freshly imported modules as you make changes.